### DISCUSSION TITLE
registers tutorial doesn't work as expected

### DISCUSSION THREAD

**Discussion by gravit22** [japaric/discovery#330] [Q&A]
When I try to cargo run --release registers I got this:
`Breakpoint 1, 0x08000242 in main ()
(gdb) n
Single stepping until exit from function main,
which has no line number information.
`

It captures input and then I need to press Ctrl+C to escape

Also this part overall not clear for me https://docs.rust-embedded.org/discovery/07-registers/rtrm.html
How do we know that we need GPIO register map 0x18 offset value?
why const GPIOE_BSRR: u32 = 0x48001018; and not 0x48001000?

**Comment by kstrckr** [comment]
> Also this part overall not clear for me https://docs.rust-embedded.org/discovery/07-registers/rtrm.html
> How do we know that we need GPIO register map 0x18 offset value?
> why const GPIOE_BSRR: u32 = 0x48001018; and not 0x48001000?

the last line on the previous page stats, "OK. But how can one find out what this register does? Time to RTRM (Read the Reference Manual)!" so the page you linked is taking you through the steps to understand what that "magic" address is doing, rather than trying to show you how to get that address yourself from scratch.

**Reply by gravit22** [reply]
Oh. Ok. Thank you. I had different goals in my had, and I thought I missed something. But thank you for clear explanation, now I know.
